modules/kubernetes-addons/external-dns/locals.tf

```diff
@@ -43,8 +43,13 @@ locals {
     irsa_iam_policies                 = concat([aws_iam_policy.external_dns.arn], var.irsa_policies)
   }
 
-  argocd_gitops_config = {
+  default_argocd_gitops_config = {
     enable             = true
     serviceAccountName = local.service_account_name
   }
+
+  argocd_gitops_config = merge(
```

Comment:
we've never done this and I think it will get quite messy with splatting out all of the values passed via `helm_config`. I believe the original intent of the bridge between Terraform and GitOps was intended to provide only the information that is created in Terraform (i.e. - IAM roles, ACM certs, etc.) to the GitOps operator. It was not designed to specify values in Terraform which are passed to the GitOps operator - those should be defined in the respective manifest repo alongside the templates/manifests that consume them

@askulkarni2 is that accurate or no?

Reply:
That may still the case with this evolution, except that we can pass dynamic values, without having to hard coded them in both git repository.

This is working, and i tested it in many different way, it's not perfect solution (it doens't allowed you to pass real values.yaml, but only root values), but it has the advantage of being dynamic (no need to synchronised both repos)

Reply:
@askulkarni2 @bryantbiggs can we validate thoses changes ?

Reply:
@askulkarni2 @bryantbiggs can we statuate on this ?
I Agrre that in the first place we can configure this values in the addon git repo, but I think it is a nice way to be able to specify some values directly from the Terraform code. This can simplify integration for customers. Please advise on this ? I also need to do the same thing for other controllers like the Argo rollout. In the Workshop there is code to activate the dashboard loadbalancer in terraform, but this didn't work because it is never passed to Argo helm chart.

Reply:
I think as far as AWS resource values are concerned (ARNs, Hosted Zone ids, OIDC providers, etc.) this is a valid use case. This is based on the assumption that platform teams usually create the cloud infrastructure required by the applications via IaC. GitOps bridge was designed for this. Unless we can come up with a better way to hand off the values from IaC state to Git state we can move forward with this approach to unblock customers. Longer term, we probably need a better approach where we can externalize the state for such dynamic resource ids and yet not break from gitops philosophy.